### PR TITLE
Remove legacy configuration system

### PR DESCRIPTION
**Pull request by williamjallen**
This PR removes the last traces of our legacy configuration system.  While somewhat substantial in size, I made an attempt to break this into incremental commits, which may be easier to review.

### CODE REVIEW

File: app/Http/Controllers/AdminController.php [hunk 1]
```diff
@@ -141,7 +138,7 @@ public function upgrade()
         @$Audit = $_POST['Audit'];
         @$ClearAudit = $_POST['Clear'];
 
-        $configFile = $config->get('CDASH_ROOT_DIR') . "/AuditReport.log";
+        $configFile = storage_path("/AuditReport.log");
```
Comment (josephsnyder) [inline app/Http/Controllers/AdminController.php:141]: This isn't the same path.  The Audit job job writes to `/cdash/app/cdash/AuditReport.log` not `/cdash/storage/AuditReport.log`

I get a 500 when clicking on the button on the maintenance page: 
image

**Review by josephsnyder — CHANGES_REQUESTED** [review]
Most of it looks reasonable, but there is an issue with the Audit job.  It should probably be changed to put it into storage, but it's not there currently.

File: app/Http/Controllers/AdminController.php (hunk 1 above)
Comment (williamjallen) [reply app/Http/Controllers/AdminController.php]: Fixed.  I'll open a separate PR later to move that file to the storage directory.

**Review by josephsnyder — APPROVED** [review]
Much better.  I can't find anything else!

**Comment by williamjallen** [comment]
I'd like @zackgalbreath to take a brief look at this before we merge it

**Review by zackgalbreath — APPROVED** [review]
LGTM, I couldn't find anything disagreeable. Thanks for caring care of this!

**Pull request merged**